engine/src/main/java/io/zeebe/engine/processing/bpmn/ProcessInstanceStateTransitionGuard.java

```diff
@@ -54,6 +55,31 @@ public boolean isValidStateTransition(final BpmnElementContext context) {
   }
 
   private Either<String, ?> checkStateTransition(final BpmnElementContext context) {
+    // migrated processors expect their state to be set via event appliers, and non migrated
+    // processors in the processor. this means that ELEMENT_COMPLETING of a migrated processor will
+    // set the element state to ELEMENT_COMPLETING on replay, but ELEMENT_COMPLETING of a non
+    // migrated will expect its state to ALREADY be ELEMENT_COMPLETING. To avoid too much
+    // complexity, when reprocessing events for non migrated processors, just set their state as
+    // expected; this could in the future be a source of error, but as its only temporary until
+    // all processors are migrated, it's OK for now
+    // TODO(npepinpe): remove as part of https://github.com/camunda-cloud/zeebe/issues/6202
+    if (!MigratedStreamProcessors.isMigrated(context.getBpmnElementType())) {
```

Comment:
It's a bit of duplication to have these rules here too, but I figured:

1. It'll be much easier to remove later
2. Introducing rules in between `case` statements with fall through is easily error prone, so I think it's worth a little duplication